src/rgw/rgw_op.h

```diff
@@ -1704,9 +1709,10 @@ static inline void rgw_get_request_metadata(CephContext *cct,
 } /* rgw_get_request_metadata */
 
 static inline void encode_delete_at_attr(ceph::real_time delete_at,
-					map<string, bufferlist>& attrs)
+					map<string, bufferlist>& attrs, bool set_delete_at = false)
 {
-  if (real_clock::is_zero(delete_at)) {
```

Comment:
It looks we started treating`0` in `delete_at` as special value saying "expiration hint revoked". Does the `expirer` part is OK with that?